File: src/models/Industry.ts

export class FeatureOption {
    constructor(
        public edition: string,
        public availability: string
    ) {}
}

export class Feature {
    constructor(
        feature: Feature
    ) { Object.assign(this, feature);}

    public fragment: string
    public isModule: boolean;
    public description: string;
    public isAvailable: boolean;
    public allowISO: string;
    public denyISO: string;
    public optionID: string;
    public options: FeatureOption[] = [];

    public customer: boolean;
    public tag: string;
    public discontinued: boolean;
    public included: boolean;
    public fromPackage: boolean
}

export class Distance {
    constructor(
        public edition: string,
        public plus: number = 0,
        public minus: number = 0
    ) {}
}

export class Weight {
    constructor(
        public edition: string,
        public min: number = 0,
        public max: number = 0
    ) {}
}

export class Industry {
    constructor(
        public name: string
    ) {}
    public industryCode: string;
    public version: string;
    public productID: string;
    public productName: string;
    public features: Feature[] = [];

    public weights: Weight[] = [];
    public clientWeight: number;
    public distances: Distance[] = [];
}

File: Models/Industry.cs
using System.Collections.Generic;

namespace Configurator.Models
 {
    public class FeatureOption
    {
        public string edition {get;set;}
        public string availability {get;set;}
    }

    public class Feature
    {
        public bool isModule {get;set;}
        public int level {get;set;}
        public string description {get;set;}
        public string fragment {get;set;}
        public bool isAvailable {get;set;}
        public string allowISO {get;set;}
        public string denyISO {get;set;}
        public string optionID {get;set;} = "";
        public List<FeatureOption> options {get;set;} = new List<FeatureOption>();
    }

    public class Industry
    {
        public string name {get; set;}
        public string industryCode {get;set;}
        public string version {get;set;}
        public string productID {get;set;}
        public string productName {get;set;}
        public List<Feature> features { get; set; } = new List<Feature>();
    }
 }